src/main/java/com/gojek/beast/sink/bq/BQClient.java

```diff
@@ -80,8 +81,21 @@ private void upsertDatasetAndTable(TableInfo tableInfo) {
     }
 
     private boolean shouldUpdateTable(TableInfo tableInfo, Table table, Schema existingSchema, Schema updatedSchema) {
+        boolean needToChangePartitionExpiry = false;
+        if (table.getDefinition().getType().equals(TableDefinition.Type.TABLE) && bqConfig.getBQTablePartitionExpiryMillis() != -1) {
+            if (shouldChangeParitionExpiryForStandardTable(table)) {
+                needToChangePartitionExpiry = true;
+            }
+        }
         return !table.getLabels().equals(tableInfo.getLabels())
-                || !BQUtils.compareBQSchemaFields(existingSchema, updatedSchema);
+                || !BQUtils.compareBQSchemaFields(existingSchema, updatedSchema)
+                || needToChangePartitionExpiry;
+    }
+
+    private boolean shouldChangeParitionExpiryForStandardTable(Table table) {
+        Long expirationMs = ((StandardTableDefinition) (table.getDefinition())).getTimePartitioning().getExpirationMs();
```

Comment:
What happens if table is not partitioned? Doesn't getTimePartitioning be null?

Reply:
yes, it will be null. I have added a check